app/src/main/res/layout/activity_animal_details.xml

```diff
@@ -10,95 +9,105 @@
+            android:id="@+id/scroll"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:background="@android:color/white"
+            android:clipToPadding="false"
+            app:layout_behavior="@string/appbar_scrolling_view_behavior">
+
+            <android.support.constraint.ConstraintLayout
                 android:layout_width="match_parent"
-                android:layout_height="300dp"
-                android:background="@color/purpleVeryLight"
-                android:contentDescription="@string/descriptor_pet_image"
-                android:elevation="4dp"
-                app:imageUrl="@{viewModel.imageUrl}"
-                app:layout_constraintTop_toBottomOf="@+id/toolbar" />
-
-            <TextView
-                android:id="@+id/animal_name"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:elevation="4dp"
-                android:paddingEnd="@dimen/standard_margin"
-                android:paddingStart="@dimen/standard_margin"
-                android:text="@{viewModel.name}"
-                android:textAppearance="@style/FontMaxHeader"
-                app:layout_constraintBottom_toTopOf="@+id/animal_breed"
-                tools:text="Mozart" />
-
-            <TextView
-                android:id="@+id/animal_breed"
+                android:layout_height="wrap_content">
+
+                <tech.pauly.findapet.shared.WrapViewPager
+                    android:id="@+id/animal_details_view_pager"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    app:adapter="@{viewModel.detailsPagerAdapter}"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toBottomOf="@+id/animal_details_tabs" />
+
+                <android.support.design.widget.TabLayout
+                    android:id="@+id/animal_details_tabs"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:tabIndicatorColor="@color/purpleDark"
+                    app:tabIndicatorHeight="2dp"
+                    app:tabMode="fixed"
+                    app:tabSelectedTextColor="@color/purpleDark"
+                    app:tabTextAppearance="@style/FontStrongBody"
+                    app:tabTextColor="@color/purpleStandard"
+                    app:viewPager="@{(animalDetailsViewPager)}" />
+            </android.support.constraint.ConstraintLayout>
+        </android.support.v4.widget.NestedScrollView>
+
+        <LinearLayout
```

Comment:
Another annoying hack. Trying to get this view behave like a FAB in the standard coordinator layout examples BUT with the view overlaying the image, not vertical aligned on the baseline of the image. Margin and padding don't work, only way I found to do it was put it inside a linear layout and align top.